tests/integration/targets/aci_netflow_monitor_policy/tasks/main.yml

```diff
@@ -0,0 +1,150 @@
+  ansible.builtin.include_tasks: ../../../../../../integration/targets/aci_cloud_provider/tasks/main.yml
+
+- name: Execute tasks only for non-cloud sites
+  when: query_cloud.current == []  # This condition will execute only non-cloud sites
+  block:  # block specifies execution of tasks within, based on conditions
+  - name: Add a new tenant
+    cisco.aci.aci_tenant: &aci_tenant_present
+      <<: *aci_info
+      tenant: ansible_tenant
+      description: Ansible tenant
+      state: present
+
+  - name: Add a Netflow Record policy
+    cisco.aci.aci_netflow_record_policy:
+      <<: *aci_info
+      tenant: ansible_tenant
+      netflow_record_policy: ansible_netflow_record_policy
+      description: Netflow Record policy for ansible_tenant tenant
+      state: present
+
+  # CREATE NETFLOW MONITOR POLICY
+  - name: Add a Netflow Monitor policy (check_mode)
+    cisco.aci.aci_netflow_monitor_policy: &aci_netflow_monitor_policy_present
+      <<: *aci_info
+      tenant: ansible_tenant
+      netflow_monitor_policy: ansible_netflow_monitor_policy_1
+      netflow_record_policy: ansible_netflow_record_policy
+      description: Netflow Monitor policy 1 for ansible_tenant tenant
+      state: present
+    check_mode: true
+    register: cm_add_netflow_monitor_policy
+
+  - name: Add a Netflow Monitor policy (normal_mode)
+    cisco.aci.aci_netflow_monitor_policy:
+      <<: *aci_netflow_monitor_policy_present
+    register: nm_add_netflow_monitor_policy
+
+  - name: Add the first Netflow Monitor policy again - testing idempotency
+    cisco.aci.aci_netflow_monitor_policy:
+      <<: *aci_netflow_monitor_policy_present
+    register: nm_add_netflow_monitor_policy_idempotency
+
+  - name: Add a second Netflow Monitor policy (normal_mode)
+    cisco.aci.aci_netflow_monitor_policy:
+      <<: *aci_info
+      tenant: ansible_tenant
+      netflow_monitor_policy: ansible_netflow_monitor_policy_2
+      description: Netflow Monitor policy 2 for ansible_tenant tenant
+      state: present
+    register: nm_add_netflow_monitor_policy_2
+
+  - name: Asserts for Netflow Monitor policy creation tasks
+    ansible.builtin.assert:
+      that:
+        - cm_add_netflow_monitor_policy is changed
+        - cm_add_netflow_monitor_policy.previous == []
+        - cm_add_netflow_monitor_policy.current == []
+        - nm_add_netflow_monitor_policy is changed
+        - nm_add_netflow_monitor_policy.current.0.netflowMonitorPol.attributes.name == "ansible_netflow_monitor_policy_1"
+        - nm_add_netflow_monitor_policy.current.0.netflowMonitorPol.children.0.netflowRsMonitorToRecord.attributes.tnNetflowRecordPolName == "ansible_netflow_record_policy"
+        - nm_add_netflow_monitor_policy_idempotency is not changed
+        - nm_add_netflow_monitor_policy_2 is changed
+        - nm_add_netflow_monitor_policy_2.previous == []
+        - nm_add_netflow_monitor_policy_2.current.0.netflowMonitorPol.attributes.name == "ansible_netflow_monitor_policy_2"
+
```

Comment:
missing test for updating attributes description an the child
missing test for removal of child or reset to "" ( if possible )

Reply:
Yes, I will add them as well, thanks.